connectors/src/connectors/microsoft/temporal/activities.ts

```diff
@@ -822,3 +822,72 @@ export async function microsoftDeletionActivity({
   }
   return new Ok(undefined);
 }
+
+export async function microsoftNodesGCActivity({
+  connectorId,
+  idCursor,
+}: {
+  connectorId: ModelId;
+  idCursor: ModelId;
+}) {
+  const connector = await ConnectorResource.fetchById(connectorId);
+  if (!connector) {
+    throw new Error(`Connector ${connectorId} not found`);
+  }
+
+  const client = await getClient(connector.connectionId);
+
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+
+  const nodes = await MicrosoftNodeResource.fetchByPaginatedIds({
+    connectorId,
+    pageSize: 100,
+    idCursor,
+  });
+
+  const lastNode = nodes.length > 0 ? nodes[nodes.length - 1] : null;
+
+  if (!lastNode) {
+    throw new Error("Unreachable: last node not found");
```

Comment:
could be the case if you have exactly 100 nodes (or anymultiple), last page will fetch all nodes and next one will fetch 0 nodes - so maybe not throw an erro in that case ?